src/tiff.rs

```diff
@@ -65,12 +92,39 @@ impl<R> TiffDecoder<R>
 }
 
 impl ImageError {
-    fn from_tiff(err: tiff::TiffError) -> ImageError {
+    fn from_tiff_decode(err: tiff::TiffError) -> ImageError {
         match err {
             tiff::TiffError::IoError(err) => ImageError::IoError(err),
-            tiff::TiffError::FormatError(desc) => ImageError::FormatError(desc.to_string()),
-            tiff::TiffError::UnsupportedError(desc) => ImageError::UnsupportedError(desc.to_string()),
-            tiff::TiffError::LimitsExceeded => ImageError::InsufficientMemory,
+            err @ tiff::TiffError::FormatError(_) => {
+                ImageError::Decoding(DecodingError::new(ImageFormat::Tiff.into(), err))
+            }
+            tiff::TiffError::UnsupportedError(desc) => {
+                ImageError::Unsupported(UnsupportedError::from_format_and_kind(
+                    ImageFormat::Tiff.into(),
+                    UnsupportedErrorKind::GenericFeature(desc.to_string()),
+                ))
+            }
+            tiff::TiffError::LimitsExceeded => {
+                ImageError::Limits(LimitError::from_kind(LimitErrorKind::InsufficientMemory))
```

Comment:
[`tiff::TiffError::LimitsExceeded`](https://docs.rs/tiff/0.4.0/tiff/enum.TiffError.html#variant.LimitsExceeded) does not say that the exceeded limit is memory.
I chosen `LimitErrorKind::InsufficientMemory` since the old code was `ImageError::InsufficientMemory`, but I'm not sure this is correct.

Reply:
The current version of `tiff` only has memory limits so it seems correct.